enhancements/machine-config/mco-cgroupsv2-support.md

```diff
@@ -0,0 +1,206 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Definitions and References
+
+- cgroup: control group and is never capitalized
+- cgroups: multiple control cgroups
+- cgroup v1: references cgroup version 1 implementation
+- cgroup v2: references cgroup version 2 implementation
+
+[Upstream Docs](https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html)
+
+## Summary
+
+Control Group v2 (cgroup v2) enablement in Kubernetes has progressed to beta
+[upstream](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2).
+The underlying runtime (cri-o) and supporting subsystems are now ready for
+customers to begin their own testing with it. Not all workloads will be
+compatible with cgroup v2, so it will *not* be enabled by default within
+OpenShift at this time.
+
+Note: This enhancement is focusing on `pure` mode cgroup v2. Mixed mode environments
+may behave differently (metrics, vpa, hpa, etc) since cgroup v1 is not
+compatible with cgroup v2.
+
+## Motivation
+
+Migrating to cgroup v2 will bring in many new features and fixes not found in
+cgroup v1. cgroup v1 is considered 'legacy' and migrating to cgroup v2 is
+necessary since RHEL8 ships with cgroup v2 on by default. (OpenShift 4.x
+currently disables cgroup v2 in favor of v1).
+
+Some features of cgroup v2 include:
+
+* IO enhancements
+* User based OOM killer
+* cgroup aware OOM killer
+
+[Kubernetes On Cgroup v2 - Video](https://www.youtube.com/watch?v=u8h0e84HxcE)
+
+### Goals
+
+- [ ] Enable cgroup v2 within the Openshift API
+- [ ] Add kernel flags to MCO to enable cgroup v2 on nodes
+
+### Non-Goals
+
+Mixed mode cgroup modes are not 100% compatible with each other. We need data
+around how cgroup v2 runs in a pure mode before we can allow mixed mode
+environments. Since, Red Hat is steering upstream cgroup v2 adoption, and we do
+not have data around the pure mode environment yet, there needs to be a platform
+to gather data from.
+
+## Proposal
+
+The option to enable cgroup v2 will have to reside in a centralized location.
+The [OpenShift Infrastructure config
+object](https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L28)
+contains information describing how a cluster functions including cloud config  
+and platform specification for each cloud. Setting the cgroup mode is an
+infrastructure setting.
+
+### API Extensions
+
+Create an additional module within openshift/api as `config/v1/types_node.go`:
+
+```go
+type CgroupMode string
+
+const (
+  CgroupMode_Empty = "" // Empty string will always use the Default
+  CgroupMode_v1 = "v1"
+  CgroupMode_v2 = "v2"
+  CgroupMode_Default = CgroupMode_v1
+)
+
+type Node struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +required
+	Spec NodeSpec `json:"spec"`
+}
+
+type NodeSpec struct {
+  CgroupMode CgroupMode `json:"cgroupMode,omitempty"`
```

Comment:
This field makes sense to me.  In the actual api PR, you'll need to fill in godoc for users.